File: interface/src/main/java/com/kgc/pinyougou/service/BrandService.java
package com.kgc.pinyougou.service;

import com.kgc.pinyougou.domain.Brand;
import com.kgc.pinyougou.utils.PageResult;

import java.util.List;
import java.util.Map;

/**
 * @Auther: 谭海亮
 * @Date: 2020/4/23 22:06
 * @Description:
 */

public interface BrandService {

    List<Map> findAllBrand();

    /**
     * @Author Admin
     * @description:  //TODO
     * @param:
     * @param pageNum   分页页数
     * @param pageSize  分页记录数
     * @return: com.kgc.pinyougou.utils.PageResult
     * @time: 2020/4/25 22:11
     */
    PageResult findPageBrand(Integer pageNum, Integer pageSize, Brand brand);

    Integer insertBrand(Brand brand);

    Brand findOne(Long id);

    Integer updateBrand(Brand brand);

    Integer deleteBrand(Long [] ids);
}

File: pinyougou-manager-web/src/main/resources/static/js/service/brandService.js
//为了达到代码的复用和维护性，将服务进行抽取
app.service("brandService", function($http) {
    this.findAllBrand=function () {
        return $http.get("../brand/findAllBrand")
    }

    this.findPageBrand=function (page,size,searchEntity) {
        return $http.post("../brand/findPageBrand?page="+page+"&size="+size,searchEntity)
    }

    this.findOne=function (id) {
        return $http.get("../brand/findOne?id="+id)
    }

    this.del=function (ids) {
        return $http.post("../brand/delBrand",ids)
    }

    this.add=function (searchEntity) {
        return $http.post("../brand/addBrand",searchEntity)
    }

    this.update=function (searchEntity) {
        return $http.post("../brand/updateBrand",searchEntity)
    }
});